File: frontend/src/api/types/seller.ts
/**
 * Seller verification types
 */

export interface SellerVerificationRequest {
  invite_code: string;
}

export interface SellerVerificationResponse {
  status: string;
  stripe_account_id: string | null;
  onboarding_url: string | null;
  message: string;
}

export interface SellerStatusResponse {
  is_seller: boolean;
  verification_status: string | null;
  charges_enabled: boolean;
  payouts_enabled: boolean;
  details_submitted: boolean;
  verified_at: string | null;
  fee_free_sales_remaining: number;
  onboarding_url: string | null;
}

export interface OnboardingLinkResponse {
  onboarding_url: string;
}

File: backend/app/schemas/seller.py
"""Seller schemas for verification requests and responses."""

from datetime import datetime

from pydantic import BaseModel, Field


class SellerVerificationRequest(BaseModel):
    """Schema for initiating seller verification via Stripe Connect onboarding."""

    invite_code: str = Field(
        ...,
        min_length=6,
        max_length=16,
        description="Invite code required to become a seller",
    )


class SellerVerificationResponse(BaseModel):
    """Schema for seller verification response."""

    status: str
    stripe_account_id: str | None = None
    onboarding_url: str | None = Field(
        default=None,
        description="One-time URL to complete Stripe Connect onboarding",
    )
    message: str


class SellerStatusResponse(BaseModel):
    """Schema for checking seller status."""

    is_seller: bool
    verification_status: str | None = None
    charges_enabled: bool = False
    payouts_enabled: bool = False
    details_submitted: bool = False
    verified_at: datetime | None = None
    onboarding_url: str | None = Field(
        default=None,
        description="One-time URL to resume Stripe Connect onboarding (pending sellers only)",
    )

    model_config = {"from_attributes": True}


class OnboardingLinkResponse(BaseModel):
    """Schema returned when a fresh Stripe onboarding link is generated."""

    onboarding_url: str
